libs/stanag4609/src/main/java/org/codice/alliance/libs/stanag4609/AsynchronousMetadataPacket.java

```diff
@@ -29,6 +35,15 @@
         // For asynchronous metadata streams, the header is supposed to be 9 bytes long. The header's
         // length field gives the number of bytes in the packet following it, so we need to skip
         // the 3 header bytes after the length field to get the true length of the payload.
-        return getPESPacketPayload(pesHeader.length - 3, ASYNCHRONOUS_PES_PACKET_HEADER_LENGTH);
+        // Some PES packets have optional fields, so get the extra byte count offset from the PES header.
+
+        if (this.pesPacketBytes.length < ASYNCHRONOUS_PES_PACKET_HEADER_LENGTH) {
+            return null;
```

Comment:
Yes, the only method that calls getKlvBytes is `final DecodedKLVMetadataPacket decodeKLV() throws KlvDecodingException`, and it handles the null.